runtime/service.go

```diff
@@ -317,6 +387,7 @@ func (s *service) Kill(ctx context.Context, req *taskAPI.KillRequest) (*ptypes.E
 	if err != nil {
 		return nil, err
 	}
+	s.cancel()
```

Comment:
I think we need proper resource cleanup on both sides:
- Wait for goroutines to finish reading/writing
- Close vsock connections if no longer needed
- Close fifos